interfaces/repo.go

```diff
@@ -516,3 +516,126 @@ func (r *Repository) securitySnippetsForSnap(snapName string, securitySystem Sec
 	}
 	return snippets, nil
 }
+
+type addSnapError struct {
```

Comment:
I've added a card to ensure this happens. As mentioned online, this is not something I've interned in my head and knowing this now will result in some more changes.